apps/trading/components/rewards-container/use-reward-row-data.spec.ts

```diff
@@ -0,0 +1,176 @@
+import type { AssetFieldsFragment } from '@vegaprotocol/assets';
+import { getRewards } from './use-reward-row-data';
+import * as Schema from '@vegaprotocol/types';
+
+const testData = {
+  rewards: [
+    {
+      rewardType: Schema.AccountType.ACCOUNT_TYPE_FEES_INFRASTRUCTURE,
+      assetId:
+        'ff3df1224e159f8619d9d66c56c21a11e10422dfa99dc45937fdff364de5e546',
+      amount: '31897424',
+    },
+    {
+      rewardType: Schema.AccountType.ACCOUNT_TYPE_FEES_INFRASTRUCTURE,
+      assetId:
+        'b340c130096819428a62e5df407fd6abe66e444b89ad64f670beb98621c9c663',
+      amount: '57',
+    },
+    {
+      rewardType: Schema.AccountType.ACCOUNT_TYPE_FEES_INFRASTRUCTURE,
+      assetId:
+        '8ba0b10971f0c4747746cd01ff05a53ae75ca91eba1d4d050b527910c983e27e',
+      amount: '5501',
+    },
+  ],
+  assets: {
+    ff3df1224e159f8619d9d66c56c21a11e10422dfa99dc45937fdff364de5e546: {
+      id: 'ff3df1224e159f8619d9d66c56c21a11e10422dfa99dc45937fdff364de5e546',
+      name: 'USD (KRW)',
+      symbol: 'USD-KRW',
+      decimals: 6,
+      quantum: '1000000',
+      source: {
+        __typename: 'ERC20',
+        contractAddress: '0xcFB5538B6235531BEbF09a04c25912e1327C5c52',
+        lifetimeLimit: '100000000000000000000',
+        withdrawThreshold: '1',
+      },
+      status: Schema.AssetStatus.STATUS_ENABLED,
```

Comment:
If you are typecasting to keep the test succint and not have to include irrelevant fields, why not remove the other irrelevant fields such as `source` or anything else that this function doesn' require

Reply:
updated. still need the cast tho